.github/workflows/build.yaml

```
@@ -71,6 +71,7 @@ jobs:
     secrets: inherit
     uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-25.06
     with:
+      matrix_filter: group_by([.ARCH, (.CUDA_VER|split(".")|map(tonumber)|.[0])]) | map(max_by(.PY_VER|split(".")|map(tonumber)))
```

```suggestion
      # build for every combination of arch and CUDA version, but only for the latest Python 
      matrix_filter: group_by([.ARCH, (.CUDA_VER|split(".")|map(tonumber)|.[0])]) | map(max_by(.PY_VER|split(".")|map(tonumber)))
```

Can you please also include a comment with the plain-language description of what this is trying to accomplish? I've found that really helpful for these filters.

We can use the same one from the corresponding `pr.yaml` line:

https://github.com/rapidsai/cugraph/blob/9e1211ace9d631bb19bae3cab5bbad554b7560e0/.github/workflows/pr.yaml#L158-L159